src/libstore/local-store.cc

```diff
@@ -136,7 +136,17 @@ LocalStore::LocalStore(
     for (auto & perUserDir : {profilesDir + "/per-user", gcRootsDir + "/per-user"}) {
         createDirs(perUserDir);
         if (!readOnly) {
-            if (chmod(perUserDir.c_str(), 0755) == -1)
+            struct stat st;
+            if (stat(perUserDir.c_str(), &st))
+                throw SysError("getting attributes of path '%1%'", perUserDir);
+
+            // Skip chmod call if the directory already has the correct permissions (0755).
+            // This is to avoid aborting when the executing user lacks permissions to change the directories permissions
+            // even if it would be no-op.
+            if ((!((st.st_mode & S_IRWXU) == S_IRWXU &&
+                (st.st_mode & S_IRGRP) == S_IRGRP &&
+                (st.st_mode & S_IROTH) == S_IROTH)) &&
```

Comment:
Can't you just do
```suggestion
            if (st.st_mode != 0755 &&
```

Reply:
We need to keep the bit shift otherwise yes, that can be shortened